### PR TITLE
[MTDSA-25054] Update mtd-identifier-lookup mapping

### PR DESCRIPTION
**Pull request by MadStu**

### CODE REVIEW

File: app/api/controllers/validators/resolvers/ResolveNino.scala [hunk 1]
```diff
@@ -34,4 +34,12 @@ object ResolveNino extends Resolver[String, Nino] {
       Invalid(List(withError(error, NinoFormatError, path)))
   }
 
+  def isValid(nino: String): Boolean = nino != null && hasValidPrefix(nino) && ninoRegex.matches(nino)
```
Comment (jeremystone) [inline app/api/controllers/validators/resolvers/ResolveNino.scala:37]: Can we update `ResolveNino` in full to ensure that `isValid` method is fully consistent with the `apply` method (i.e. `apply` uses `isValid`). Otherwise we run the risk of having anomalous edge cases.

You should also update `ResolveNinoSpec`.

File: test/utils/ErrorHandlerSpec.scala [hunk 2]
```diff
@@ -103,9 +103,9 @@ class ErrorHandlerSpec extends UnitSpec with GuiceOneAppPerSuite {
     "return 401 with error body" when {
       "unauthorised and header is supplied" in new Test() {
         val result: Future[Result] = handler.onClientError(requestHeader, UNAUTHORIZED, "test")
-        status(result) shouldBe UNAUTHORIZED
+        status(result) shouldBe ClientOrAgentNotAuthorisedError.httpStatus
```
Comment (jeremystone) [inline test/utils/ErrorHandlerSpec.scala:106]: Best be explicit in tests about what you want the behaviour to be. It's not that we want the status to be _whatever_ is in `ClientOrAgentNotAuthorisedError.httpStatus`: we want it to be `UNAUTHORIZED`.
So this line does not need to change as `ErrorHandler` behaviour should remain unchanged. (You will need to fix up `ErrorHandler` accordingly.)

File: it/v5/AuthISpec.scala [hunk 3]
```diff
@@ -62,22 +62,23 @@ class AuthISpec extends IntegrationBaseSpec {
...
           AuditStub.audit()
-          MtdIdLookupStub.internalServerError(nino)
+          MtdIdLookupStub.error(nino, Status.INTERNAL_SERVER_ERROR)
         }
 
         val response: WSResponse = await(request().post(Json.parse(requestJson)))
         response.status shouldBe Status.INTERNAL_SERVER_ERROR
       }
     }
+  }
```
Comment (jeremystone) [inline it/v5/AuthISpec.scala:79]: This needs to include the "MTD ID lookup fails with a 403" case as this is the scenario that this ticket should fix. (It might be instructive to add the test to the code base without your changes and watch it fail.)

**Review by jeremystone — APPROVED** [review]
